### PR TITLE
Upgrade SVGO to version 2

### LINKED ISSUE #5078: Migrate SVGO to v2
It seems that [SVGO maintenance is back](https://github.com/svg/svgo/releases/tag/v2.0.0)! :tada: So will be worth to migrate to the new version and stay tuned with changes.

The configuration must be handled in `svgo.config.js` now.

### PR DESCRIPTION
**Pull request by mondeja**
Closes #5078.

SVGO development comes back with the [v2 release](https://github.com/svg/svgo/releases/tag/v2.0.0), with the most important breaking change of the configuration definition in a `svgo.config.js` file. This pull migrates the previous configuration to the new file and updates the dependency in `package.json`.

### CODE REVIEW

**Review by ericcornelissen — CHANGES_REQUESTED** [review]
Thanks for picking this one up @mondeja :tada: 

One minor thing, you forgot to update [the `npm run svgo` script definition](https://github.com/simple-icons/simple-icons/blob/c4a4c7893845b91bcfcd0a306333c98808043992/package.json#L45). Following the [SVGO v2 release notes](https://github.com/svg/svgo/releases/tag/v2.0.0), we may omit specifying the configuration file (as `svgo.config.js`) is the default. Personally i would prefer we keep it explicit, but either way works for me :slightly_smiling_face: 

Another small point, I believe it is also supported to just have the name of the plugin in the array of plugins, I would prefer that for the _"Explicitly enable everything else"_ list (if it works, I can't test it right now).

**Comment by mondeja** [comment]
> Another small point, I believe it is also supported to just have the name of the plugin in the array of plugins, I would prefer that for the "Explicitly enable everything else" list (if it works, I can't test it right now).

Tested and works :+1:

File: svgo.config.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+      params: {
+        // 3 decimals of precision in floating point numbers
+        floatPrecision: 3,
+        // Some editors (eg. Adobe Illustrator and Sketch) does not parse flags
+        // without space wrapping
+        noSpaceAfterFlags: false,
+      }
+    },
+
+    // Sort attributes on the <SVG>
```
Comment (ericcornelissen) [inline svgo.config.js:17]:
```suggestion
    // Sort the attributes on the <svg> tag
```

File: svgo.config.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    {
+      name: 'sortAttrs',
+      params: {
+        order: ['role', 'viewBox'],
+        xmlnsOrder: 'end',
+      }
+    },
+
+    // Convert basic shapes (such as <circle>) to <path>, including <arc>
```
Comment (ericcornelissen) [inline svgo.config.js:26]:
```suggestion
    // Convert all basic shapes (such as <circle>) to <path>
```

File: svgo.config.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    },
+
+    // Keep the <title> tag
+    {
+      name: 'removeTitle',
+      active: false,
+    },
+
+    // Don't remove the role="img" attribute and automatically
+    // add it to the SVG if it's not
```
Comment (ericcornelissen) [inline svgo.config.js:50]:
```suggestion
    // Keep the role="img" attribute and automatically add it
    // to the <svg> tag if it's not there already
```

File: svgo.config.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    {
+      name: 'addAttributesToSVGElement',
+      params: {
+        attributes: [
+          {role: 'img'},
+        ]
+      }
+    },
+
+    // Keep the 'role' attribute, if is already defined
```
Comment (ericcornelissen) [inline svgo.config.js:60]:
```suggestion
    // Keep the 'role' attribute if it's already defined
```

File: svgo.config.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    {
+      name: 'removeUnknownsAndDefaults',
+      params: {
+        keepRoleAttr: true,
+      }
+    },
+
+    // Remove all attributes except 'role', 'viewBox' and 'xmlns' of <svg>s
```
Comment (ericcornelissen) [inline svgo.config.js:68]:
```suggestion
    // Remove all attributes except 'role', 'viewBox', and 'xmlns' from
    // <svg> tags
```

File: svgo.config.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+module.exports = {
+  multipass: true,
+  plugins: extendDefaultPlugins([
+    {
+      name: 'convertPathData',
+      params: {
+        // 3 decimals of precision in floating point numbers
+        floatPrecision: 3,
+        // Some editors (eg. Adobe Illustrator and Sketch) does not parse flags
+        // without space wrapping
```
Comment (ericcornelissen) [inline svgo.config.js:12]: Minor grammatical mistake, but still very much understandable

```suggestion
        // Some editors (e.g. Adobe Illustrator and Sketch) cannot parse flags
        // without space wrapping
```

**Review by ericcornelissen — APPROVED** [review]
> Tested and it works :+1:

Awesome :tada: In theory this is good to merge now, thanks @mondeja 

Ideally, we would improve some of the other comments as well (for clarity and consistency, as per my suggestions below) now that we're editing the file anyway. And one other nitpick is that you used trailing commas everywhere except for the `params: { }` objects. If you want to change these thing @mondeja that would be nice, but feel free to merge this as is :smiley:

**Pull request merged**